.github/workflows/deploy-and-run-cypress.yml

```diff
@@ -0,0 +1,118 @@
+name: Deploy and run test with API
+# Build and deploy test instance for every pull request
+
+on:
+  workflow_dispatch:
+    inputs:
+      api_url:
+        required: true
+        description: 'Build dashboard and run test with this API url'
+        type: string
+      tags:
+        required: true
+        description: 'Build dashboard and run test with this API url'
+        type: string
+        default: "@stable"
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    outputs:
+      base_URL: ${{ steps.set-domain.outputs.domain }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: rlespinasse/github-slug-action@3.1.0
+
+      - name: Start deployment
+        uses: bobheadxi/deployments@v0.4.2
+        id: deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: ${{ env.GITHUB_HEAD_REF_SLUG_URL }}
+          ref: ${{ github.head_ref }}
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-qa-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-qa-${{ env.cache-name }}-
+            ${{ runner.os }}-qa-
+            ${{ runner.os }}-
+
+      - name: Install deps
+        run: |
+          npm ci
+
+      - name: Run build
+        env:
+          API_URI: ${{github.event.inputs.api_url}}
+          APP_MOUNT_URI: /
+          STATIC_URL: /
+          IS_CLOUD_INSTANCE: true
+        run: |
+          npm run build
+
+      - name: Set domain
+        id: set-domain
+        # Set test instance domain based on branch name slug
+        run: |
+          echo "::set-output name=domain::${{ env.GITHUB_HEAD_REF_SLUG_URL }}.dashboard.saleor.rocks"
```

Comment:
This line is identical as the one here:
https://github.com/saleor/saleor-dashboard/blob/1002a11f41116b25045edf9a7dd525f748aee5e1/.github/workflows/test-env-deploy.yml#L84

Please add some prefix in order to avoid overwriting existing test env